org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java

```diff
@@ -183,11 +185,17 @@ public synchronized void closeSonarConsole() {
 
   public synchronized ListenerFactory listenerFactory() {
     if (listenerFactory == null) {
-      listenerFactory = () -> notification -> Display.getDefault().asyncExec(() -> {
-        SonarQubeNotificationPopup popup = new SonarQubeNotificationPopup(Display.getCurrent(), notification);
-        popup.create();
-        popup.open();
-      });
+      // don't replace the anon class with lambda, because then the factory's "create" will always return the same listener instance
+      listenerFactory = () -> new SonarQubeNotificationListener() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1604)